pkg/provisioner/ironic/ironic.go

```diff
@@ -2125,3 +2125,33 @@ func (p *ironicProvisioner) RemoveBMCEventSubscriptionForNode(subscription metal
 	}
 	return operationComplete()
 }
+
+// Get DataImage Details
+// Fetches the VirtualMedia details of the BareMetalHost and updates DataImage.
+func (p *ironicProvisioner) GetDataImageStatus(dataImage *metal3api.DataImage) (err error) {
```

Comment:
Provisioners should not handle API objects directly, it's the responsibility of controllers.